src/main/java/com/fruits/ecommerce/repository/CategoryRepository.java

```diff
@@ -0,0 +1,10 @@
+package com.fruits.ecommerce.repository;
+
+import com.fruits.ecommerce.models.entities.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CategoryRepository extends JpaRepository<Category,Long> {
+    boolean existsByName(String name);
```

Comment:
Would you provide generated SQL for this method, I believe it might not be optimized for performance

Reply:
You are right; I will change the code as per the improvement you suggested.